docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+# Table of Content
+
+- [Client facade](#client-facade)
+- [Controller](#controller)
+- [Dependency Provider](#dependency-provider)
+- [Entity](#entity)
+- [Entity Manager](#entity-manager)
+- [Facade](#facade)
+- [Facade design pattern](#facade-design-pattern)
+- [Factory](#factory)
+- [Gateway Controller](#gateway-controller)
+- [Layout](#layout)
+- [Mapper / Expander / Hydrator](#mapper--expander--hydrator)
+- [Model](#model)
+- [Module Configurations](#module-configurations)
+- [Navigation.XML](#navigationxml)
+- [Permission Plugin](#permission-plugin)
+- [Persistence Schema](#persistence-schema)
+- [Provider / Router](#provider--router)
+- [Query Container facade](#query-container-facade)
+- [Query Object](#query-object)
+- [Repository](#repository)
+- [Service facade](#service-facade)
+- [Theme](#theme)
+- [Transfer Object](#transfer-object)
+- [Widget](#widget)
+- [Zed Stub](#zed-stub)
+- [Bridge](#bridge)
+- [Plugin](#plugin)
+- [Plugin Interface](#plugin-interface)
+
+# Documentation structure
+
+## Development Use-Cases
+Understanding the development scenarios in which Spryker Commerce Operating System (SCOS) can be utilized is crucial for maximizing its potential. We have outlined specific behaviors and guidelines tailored to different use-cases:
+
+- **Project Development**: If you are developing a project, you will need to adhere to specific project development guidelines to ensure smooth integration.
+- **Boilerplate and Accelerator Contribution**: Contributing to boilerplates or accelerators requires additional considerations. We provide guidelines specific to this use-case to streamline your contributions.
+- **Module Contribution**: For those contributing to modules, there are specialized rules to follow within the module folders. This ensures consistency and compatibility across SCOS.
+
+## Directive Classification
+Throughout this documentation, you will encounter two types of directives:
+- **Convention**: These are mandatory requirements that contributors must adhere to in order to enable specific SCOS features or ensure proper application responses.
+- **Guideline**: While not mandatory, following these guidelines is highly recommended. Doing so promotes long-term code maintainability and facilitates smoother development processes.
+
+# Applications
+Spryker utilises application layers into applications to enable constructing the necessary application architecture for the specific business requirements to provide a quick project start, and long-term maintainability.
+- Backend applications (eg: Zed, Backend API, Backend Gateway, Backoffice, GlueBackend, MerchantPortal) typically use the Zed-, Glue-, Client-, Service-, and Shared application layers.
```

Comment:
I read your conversation, but I would still name application, not by the folder name, but by their name. Also mb we can put references to pages that are describing those applications or at least add additional context. E.g. new Glue Infrastructure page or decoupled application for Zed

Reply:
could you please propose it in a diff?